### PR TITLE
Order features implementation

### LINKED ISSUE #25: Order operations
Allow CRUD for order operations.

### LINKED ISSUE #64: Tests for items
- Implement tests for items

### LINKED ISSUE #67: Loading list component
There are plenty of lists in the app that use the same logic for loading indicators.
This should be extracted into a reusable component to avoid code-duplication

### LINKED ISSUE #70: Allow change company while viewing items
User should be allowed to change company while viewing items.
Purpose:
    - This will serve when "buying" items 
       - The user will be able to navigate quickly between companies to add or remove items         
          without having to reset the navigation back to dashboard

### Linked pull requests
#66

### LINKED ISSUE #71: Fab icon for changing company
### Proposed implementation
Add a fab icon to open a bottom sheet, on the bottom sheet the user should be able to select a company that will refresh the page

_Originally posted by @HarryPi in https://github.com/Quantium-Nonsense/ProjectOmega/issues/70#issuecomment-612121136_

### LINKED ISSUE #73: Company items effects testing lacking
There is not good enough testing coverage for the new functionality of company items effect

### LINKED ISSUE #76: Order options
When the user hits the order cart a menu should  be presented.
The menu should be able to perform the following:

- Clear all items in order.
- Proceed to review and place order.

### PR DESCRIPTION
**Pull request by HarryPi**
## New features
- Moved toolbar into its own component for easier maintenance 93fca1efeaee7759338f032c90abfa74739f8c2f
  - Added cart to toolbar that will expose further functionality
- Added a button at order page for user to add items of any company from there if desire so.
## Known Bugs
- ~~No burger menu when viewing company items~~ Failed to reproduce

## Testing
- Testing for company items completed.
- Added further tests for all effects related to the app state 16b503c2a877328f372e1a06091b6fda5fa3973b
### In progress testing
- ~~Company testing lacking greatly see #73~~ see 16b503c2a877328f372e1a06091b6fda5fa3973b

## Minor changes
- Restricted dummy items price below £100

## Linked issues
- Closes #25   Closes #70   Closes #76
- Closes #64   Closes #71
- Closes #67   Closes #73

### CODE REVIEW

File: RepClient/src/app/order/store/order.effects.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    updatedOrder = [..._.cloneDeep(currentState.items)]; // Never mutate state
+
+    if (currentState.items) {
+      const lookupItem: OrderItemModel = updatedOrder.find(i => i.id === item.id); // Check if our item is there
+
+      if (lookupItem) {
+        lookupItem.quantity -= 1;
+        if (lookupItem.quantity <= 0) { // if quantity 0 or less
+          updatedOrder.splice(updatedOrder.indexOf(lookupItem), 1); // Remove item
+        }
```
Comment (markscamilleri) [inline RepClient/src/app/order/store/order.effects.ts:63]: What if quantity, was somehow already `<=0`?

File: RepClient/src/app/shared/component/add-remove-item/add-remove-item.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      fromOrder.selectItems,
+      []
+    );
+
+    fixture.detectChanges();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
Comment (markscamilleri) [inline RepClient/src/app/shared/component/add-remove-item/add-remove-item.component.spec.ts:40]: This test alone is not enough

File: RepClient/src/app/shared/component/header/header.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    fixture = TestBed.createComponent(HeaderComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (markscamilleri) [inline RepClient/src/app/shared/component/header/header.component.spec.ts:32]: This test alone is not enough

Comment (HarryPi) [reply RepClient/src/app/shared/component/header/header.component.spec.ts:32]: Tests where deemed not appropriate for this pull request and would be added as a separate pull request. However that was put on hold for now to aid with the Admin front end as it is more pressing to have it finished.

File: RepClient/src/app/shared/component/add-remove-item/add-remove-item.component.spec.ts (hunk 2 above)
Comment (HarryPi) [reply RepClient/src/app/shared/component/add-remove-item/add-remove-item.component.spec.ts:40]: Tests where deemed not appropriate for this pull request and would be added as a separate pull request. However that was put on hold for now to aid with the Admin front end as it is more pressing to have it finished.

File: RepClient/src/app/order/order.page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    fixture = TestBed.createComponent(OrderPage);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
[the remark this answers is not in the document]
Comment (HarryPi) [reply RepClient/src/app/order/order.page.spec.ts:32]: Tests where deemed not appropriate for this pull request and would be added as a separate pull request. However that was put on hold for now to aid with the Admin front end as it is more pressing to have it finished.

File: RepClient/src/app/order/store/order.effects.ts (hunk 1 above)
Comment (HarryPi) [reply RepClient/src/app/order/store/order.effects.ts:63]: Then the item would be removed regardless as it fulfils the condition?

Comment (markscamilleri) [reply RepClient/src/app/order/store/order.effects.ts:63]: Nvm yeah, idk what I was thinking

**Pull request merged**